[REVIEW] Testing Procedure v0.1

narvcharv opened the pull request:
Closes #117

JLefebvre55 committed:
Setup

narvcharv committed:
early draft testing plan + control points

basically a frame, need to ask questions then can fill it out properly

JLefebvre55 commented:
Review round 1:
- [ ] General Testing Plan - right ideas?
- [ ] HACCP Critical Points - right ones? any missing?

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
+% Suggestion: Double-blind volunteer study
+Tested via blind studies where participants are divided into two groups and given either control outputs (i.e. established commercial product) or test outputs (i.e. produced by PeaPod). Participants will rate outputs on 4 criteria (appearance, aroma, flavour, and texture) on a 9-point scale.
+
+Blind studies are eminent in consumer testing as they allow researchers to get a completely unbiased dataset. Special care needs to be taken when presenting, preparing, and collecting samples for testing to ensure researchers do not influence results. Ideally, resources will permit a double-blind study where researchers hire an outside entity to conduct the test and return results with generic la ...
```
Cite existing studies that use this, or some other academic basis for this method. also derive a more detailed methodology from that

JLefebvre55 replied:
Hedonic scale?

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
 % Biohazards - total aerobic plate count, ATP testing of food contact surfaces
 % TODO: What do those^ mean? How do we test them?
+\subsubsection{Chemical Hazards}
+As per the parts manifest !LINK, all components involved in the unit are clear of toxins and heavy metals by composition.
```
How did we avoid "typical" pitfalls (i.e. leaded vs lead-free solder, user chooses composition of nutrient solutions, all parts in contact with plant and water supply are food-safe (brass, petg, teflon tape, etc.))

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
+As per the parts manifest !LINK, all components involved in the unit are clear of toxins and heavy metals by composition.
+
+\subsubsection{Biological Hazards}
+Aerobic Plate Count (APC) testing to be done with the Conventional Plate Count Method outlined in the FDA's \href{https://www.fda.gov/food/laboratory-methods-food/bam-chapter-3-aerobic-plate-count}{BAM Chapter 3: Aerobic Plate Count}. This is selected over the Spiral Plate Method as it is inexpensive and uses many household materials. The goal of APC testing is to indicate the bacterial populatio ...
```
Combine details from standard (materials, duration, time between tests, number of tests, etc.) with real-life academic use-cases (paper methodologies)

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
+\subsubsection{Biological Hazards}
+Aerobic Plate Count (APC) testing to be done with the Conventional Plate Count Method outlined in the FDA's \href{https://www.fda.gov/food/laboratory-methods-food/bam-chapter-3-aerobic-plate-count}{BAM Chapter 3: Aerobic Plate Count}. This is selected over the Spiral Plate Method as it is inexpensive and uses many household materials. The goal of APC testing is to indicate the bacterial populatio ...
+
+ATP testing to be done using !CITE (lots of stuff about methods but no standards? look at requirements more)
```
Specific method? Standard?

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
 \subsection{Safety of Outputs}
 % SPECIFICALLY food outputs
 % Biohazards - total aerobic count, specific pathogens (enterobacteriaceae, salmonella, yeasts, molds, E. coli, Listeria, etc.)
+APC testing conducted on samples as outlined above to ensure bacterial population below 20 000 CFU/g per STD-3001.
```
Combine details from standard (materials, duration, time between tests, number of tests, etc.) with real-life academic use-cases (paper methodologies)

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
 % Biohazards - total aerobic count, specific pathogens (enterobacteriaceae, salmonella, yeasts, molds, E. coli, Listeria, etc.)
+APC testing conducted on samples as outlined above to ensure bacterial population below 20 000 CFU/g per STD-3001.
+
+Critical pathogens to be tested for individually:
```
E. coli and listeria should be 0 CFU/g (our tech does not need it for production, ergo we should eliminate it)

JLefebvre55 replied:
Side note: research UV LED spectrum and intensity vs disinfection rate and/or duration-to-disinfect? does this change for different pathogens? is there a "catch-all"?

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
 
 \subsection{Reliability and Stability of Outputs}
 % Biohazards - total aerobic plate count, enterobacteriaceae, yeasts, molds
+%is this not the same as safety of outputs?
```
Time-related with regard to outputs. I.e. how long does organic pesticide-free hydroponically-grown fresh produce last? what is the APC of various pathogens at different "ages"?

JLefebvre55 replied:
Side note: implementing "overhead misting" or some kind of deployment mechanism for X-icides?

JLefebvre55 commented on docs/tex/TestingProcedure.tex:
```diff
@@ -7,24 +7,54 @@ \subsection{Acceptability}
-% Incl. sample collection (size, timing, quantity), packaging, shipping
\ No newline at end of file
+% Incl. sample collection (size, timing, quantity), packaging, shipping
+The number of days required for sample collection is entirely dependnent on what sample is being produced. For one-time growth products, such as carrots or lettuce, the days required is exactly the time to harvest of the plant. Size of collection is dependent on how many units are run at the same time. For plants that produce products multiple times, such as beans or tomatoes, samples should be c ...
```
Sample collections at different plant ages? i.e. young basil vs fully-grown basil for different quality analyses

JLefebvre55 commented on docs/tex/HACCP.tex:
```diff
@@ -5,28 +5,55 @@ \section{Hazard Analysis and Critical Control Point (HACCP) Plan}
 % Incl. flowchart
 
+
 \subsection{Critical Points}
```
New definitional understanding: a critical point is a minimally-complex system (i.e. cannot be realistically/usefully broken down into smaller systems), i.e. a single component that accomplishes a single function that would otherwise be FAILING.

JLefebvre55 replied:
Start at the macro-system level (aeroponics, lighting, etc.) and break down from there. List the ones you know AND are included in the solution overview

JLefebvre55 replied:
I think what you have here is still good and stuff like it should be included, just add the above to it

JLefebvre55 commented on docs/tex/HACCP.tex:
```diff
@@ -5,28 +5,55 @@ \section{Hazard Analysis and Critical Control Point (HACCP) Plan}
-% Monitoring procedures
-% Deviation procedures
-% Associated documents
+\textbf{Hazard Description}
```
What is the variable being controlled? What is the function of the control point?

JLefebvre55 commented on docs/tex/HACCP.tex:
```diff
@@ -5,28 +5,55 @@ \section{Hazard Analysis and Critical Control Point (HACCP) Plan}
+% How do we know the state of the CCP
+
+\textbf{Deviation Procedures}
+% aka corrective actions
```
And fail-safe

JLefebvre55 commented on docs/tex/HACCP.tex:
```diff
@@ -5,28 +5,55 @@ \section{Hazard Analysis and Critical Control Point (HACCP) Plan}
 \subsubsection{Critical Point ...}
 
 \subsection{Standard Test Record}
+% TODO: What exactly is this?
```
Research this - how do the following subheadings pan out?

JLefebvre55 reviewed (changes requested):
Good stuff. Address the feedback and let's get er dun yeeeeeee

JLefebvre55 committed:
Merge branch 'staging' into docs-testingplan

JLefebvre55 committed:
Restructuring

- Removed solution overview (too similar to progress report, will likely refactor back after dsfc)
* Moved, combined subsystem doc fragments

JLefebvre55 committed:
Updated Subsystem Design Details

* Updated details for each subsystem
+ Added old bib references
+ Added part specifics
* Fixed all \ref and \label

JLefebvre55 committed:
Added notes

narvcharv committed:
Update Acceptability in TestingProcedure.tex

narvcharv committed:
Update Acceptability in TestingProcedure.tex

JonasMarshall committed:
Update TestingProcedure.tex

Added info under safety of outputs and reliability and stability of outputs

narvcharv committed:
Update Safety of Process in TestingProcedure.tex

narvcharv committed:
Merge branch 'docs-testingplan' of https://github.com/PeaPodTechnologies/PeaPod into docs-testingplan

narvcharv committed:
Update Chemical Hazards in TestingProcedure.tex

narvcharv committed:
Update Resource Outputs in TestingProcedure.tex

JLefebvre55 committed:
Restructured Doc, Added Graphics

+ Solenoid lock

JLefebvre55 committed:
Process Description

* Improved equations
+ Added parts
* improved layout, page breaks

JLefebvre55 committed:
Prototype Build Status

JLefebvre55 committed:
Restructuring

JLefebvre55 committed:
Title

JLefebvre55 committed:
Merge branch 'docs-progressreport' into docs-testingplan

JLefebvre55 committed:
System Description

removed animation folder
moved design report out of archive

JLefebvre55 committed:
Materials: Automation

narvcharv committed:
Update Sample Collection for Acceptability

narvcharv committed:
Update Sample Collection for Safety of Outputs

narvcharv committed:
Update Sample Collection for Reliability and Stability

JLefebvre55 committed:
Merge branch 'staging' into docs-testingplan

JLefebvre55 committed:
Materials: Housing

JLefebvre55 committed:
Materials: Aeroponics

split housing (shell, grow and lighting trays)
table formatting

JLefebvre55 committed:
Materials: Thermoregulation

JLefebvre55 committed:
Materials: Thermoregulation pt. 2

JLefebvre55 committed:
Materials: Humidification, Dehumidification, Gas

JLefebvre55 committed:
Merge branch 'staging' into docs-testingplan

narvcharv committed:
Update TestingPlan.tex

Small addition to safety

JLefebvre55 committed:
Materials: Lighting

JLefebvre55 committed:
Merge branch 'docs-testingplan' of https://github.com/PeaPodTechnologies/PeaPod into docs-testingplan

JLefebvre55 committed:
Added Desiccant, Air Filters

JLefebvre55 committed:
Materials: Inputs and Outputs

JLefebvre55 committed:
Materials: Maintenance and Cleaning

JLefebvre55 committed:
Cleanup

Formatting, char counts, references

narvcharv committed:
Update Testing Plan - By-Product Safety

narvcharv committed:
Update Testing Plan - incorporate feedback

JLefebvre55 committed:
Hazard Analysis: Crew Contact

JLefebvre55 committed:
Cleanup

JLefebvre55 committed:
Minor Fixes

narvcharv committed:
Hazard Analysis: Water Supply

narvcharv committed:
Hazard Analysis: Nutrient Supply

and some fixes for water supply

narvcharv committed:
Hazard Analysis: Seeds

narvcharv committed:
Hazard Analysis: Testing

JLefebvre55 committed:
Minor Changes

JLefebvre55 committed:
Convert to Markdown

JLefebvre55 committed:
Testing Plan & HACCP v1

JLefebvre55 commented:
Finished, merging manually
